govcd/org.go

```diff
@@ -48,6 +48,62 @@ func NewAdminOrg(c *Client) *AdminOrg {
 	}
 }
 
+// Generic function that takes in a pointer to a Org or Admin org as an
+// entity and refreshes it by getting the current state from the
+// vcd rest api. Users should use this function after creating,
+// updating, deleting resources under this org. Since the we use
+// org links to get elements within the org, users must refresh
+// after performing an operation that can change the links. Otherwise
+// users might get stale metadata, and for users trying to delete an org
+// may run into issues deleting everything.
```

Comment:
Hmm.  This does not seem better than the previous implementation which had two copy/paste functions.  It's still effectively two versions because there are a bunch of if statements to break up the control flow--this seems more complex than what you had before. 

Unless there's a way in Go to paper over the differences between the structs and allow a single path for both types, I would revert this back to the previous implementation.